circuit_maintenance_parser/processor.py

```diff
@@ -117,16 +117,27 @@ class CombinedProcessor(GenericProcessor):
 
     def process_hook(self, maintenances_extracted_data, maintenances_data):
         """All the parsers contribute with a subset of data that is extended."""
-        # We only expect one data object from these parsers
         if len(maintenances_extracted_data) == 1:
             self.combined_maintenance_data.update(maintenances_extracted_data[0])
         else:
-            raise ProcessorError(f"Unexpected data retrieved from parser: {maintenances_extracted_data}")
+            maintenances_data.extend(maintenances_extracted_data)
 
     def post_process_hook(self, maintenances_data):
         """After processing all the parsers, we try to combine all the data together."""
         self.extend_processor_data(self.combined_maintenance_data)
-        try:
-            maintenances_data.append(Maintenance(**self.combined_maintenance_data))
-        except ValidationError as exc:
-            raise ProcessorError("Not enough information available to create a Maintenance notification.") from exc
+        if maintenances_data:
+            maintenances = maintenances_data.copy()
+            maintenances_data.clear()
+            for maintenance in maintenances:
+                try:
+                    combind_data = {**self.combined_maintenance_data, **maintenance}
+                    maintenances_data.append(Maintenance(**combind_data))
+                except ValidationError as exc:
+                    raise ProcessorError(
+                        "Not enough information available to create a Maintenance notification."
+                    ) from exc
+        else:
```

Comment:
what if you do something like:

```
if not maintenances_data:
    maintenance_data = [{}]

maintenances = maintenances_data.copy()
maintenances_data.clear()
...
```

so, you don't need the last try/except 😄

Reply:
Because your re-initalizing `maintenance_data` the value cannot be passed back out of the function.
So ive implemented the code before
``` python
        if not maintenances_data:
            maintenances = [{}]
        else:
            maintenances = maintenances_data.copy()
            maintenances_data.clear()
```